boards: arm: add initial support for NUCLEO-L073RZ board

tagunil opened the pull request:
Tested on hello_world, blinky and button samples.

tagunil committed:
arch: arm: soc: stm32l0: fix indentation and comments in Kconfigs

Fix indentation and small comment typos in Kconfigs.

Signed-off-by: Ilya Tagunov <tagunil@gmail.com>

tagunil committed:
arch: arm: soc: stm32l0: add STM32L073xZ SOC

Add initial support for STM32L073xZ SOC which is not very different
from already supported STM32L072xZ.

Signed-off-by: Ilya Tagunov <tagunil@gmail.com>

tagunil committed:
boards: arm: add NUCLEO-L073RZ board

Add initial support for NUCLEO-L073RZ board.
This is a typical Nucleo-64 board, so most files are reused
from already supported Nucleo boards with small changes.

Signed-off-by: Ilya Tagunov <tagunil@gmail.com>

ydamigos commented on boards/arm/nucleo_l073rz/nucleo_l073rz.dts:
```diff
@@ -0,0 +1,32 @@
+ */
+
+/dts-v1/;
+#include <st/stm32l073.dtsi>
```
I couldn't find stm32l073.dtsi file. Is it missing?

ydamigos replied:
My bad. I found it.

tagunil replied:
No, it is added by ebb91ac2a39b76b60e2d892974c09248cae8a240.

ydamigos reviewed (changes requested):
Looks good but I couldn't find stm32l073.dtsi file.

tagunil commented on boards/arm/nucleo_l073rz/board.h:
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2017 RnDity Sp. z o.o.
```
Done.

dbkinder commented on boards/arm/nucleo_l073rz/doc/nucleol073rz.rst:
```diff
@@ -0,0 +1,161 @@
+
+- STM32 microcontroller in QFP64 package
+- Two types of extension resources:
+    - Arduino* Uno V3 connectivity
```
need a blank line before a list (including sub-lists), so add a blank line here.

dbkinder commented on boards/arm/nucleo_l073rz/doc/nucleol073rz.rst:
```diff
@@ -0,0 +1,161 @@
+    - ST morpho extension pin headers for full access to all STM32 I/Os
+- ARM* mbed*
+- On-board ST-LINK/V2-1 debugger/programmer with SWD connector:
+    - Selection-mode switch to use the kit as a standalone ST-LINK/V2-1
```
add blank line here

dbkinder commented on boards/arm/nucleo_l073rz/doc/nucleol073rz.rst:
```diff
@@ -0,0 +1,161 @@
+- On-board ST-LINK/V2-1 debugger/programmer with SWD connector:
+    - Selection-mode switch to use the kit as a standalone ST-LINK/V2-1
+- Flexible board power supply:
+    - USB VBUS or external source (3.3V, 5V, 7 - 12V)
```
blank line

dbkinder commented on boards/arm/nucleo_l073rz/doc/nucleol073rz.rst:
```diff
@@ -0,0 +1,161 @@
+    - USB VBUS or external source (3.3V, 5V, 7 - 12V)
+    - Power management access point
+- Three LEDs:
+    - USB communication (LD1), user LED (LD2), power LED (LD3)
```
add a blank line

dbkinder commented on boards/arm/nucleo_l073rz/doc/nucleol073rz.rst:
```diff
@@ -0,0 +1,161 @@
+    - USB communication (LD1), user LED (LD2), power LED (LD3)
+- Two push-buttons: USER and RESET
+- USB re-enumeration capability. Three different interfaces supported on USB:
+    - Virtual COM port
```
blank line

dbkinder commented on boards/arm/nucleo_l073rz/doc/nucleol073rz.rst:
```diff
@@ -0,0 +1,161 @@
+    - Mass storage
+    - Debug port
+- Support of wide choice of Integrated Development Environments (IDEs) including:
+    - IAR
```
blank line

dbkinder commented on boards/arm/nucleo_l073rz/doc/nucleol073rz.rst:
```diff
@@ -0,0 +1,161 @@
+    - GCC-based IDEs
+
+More information about STM32L073RZ can be found here:
+       - `STM32L0x3 reference manual`_
```
better to write this as:

    More information about STM32L073RZ can be found in the
    `STM32L0x3 reference manual`_.

dbkinder commented on boards/arm/nucleo_l073rz/doc/nucleol073rz.rst:
```diff
@@ -0,0 +1,161 @@
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+- UART_1_TX : PB6
```
add a blank line before a list

dbkinder reviewed (changes requested):
Thanks for the documentation.  A few formatting changes recommended.

tagunil commented:
@dbkinder, I think I've addressed your comments. Should I apply same changes to other Nucleos documentation in a separate PR? These boards are very similar, so it's mostly a copy-paste from each other.

dbkinder commented:
oh dear.  Looks like those other doc files snuck by without notice that the indentation isn't right.  If you're in the mood to fix those that would be good.  But I can add it to my list of things to fix otherwise.  Let's start by getting this doc right.

dbkinder commented on boards/arm/nucleo_l073rz/doc/nucleol073rz.rst:
```diff
@@ -0,0 +1,168 @@
+- STM32 microcontroller in QFP64 package
+- Two types of extension resources:
+
+    - Arduino* Uno V3 connectivity
```
Sorry, ReST is picky about indentation.  I just noticed too that these nested lists should have the indentation so the ``-`` lines up with the first letter of the parent list item or preceding sentence (otherwise it adds an extra blank line in the rendered output):

    - Two types of extension resources:

      - Arduino ...

dbkinder commented on boards/arm/nucleo_l073rz/doc/nucleol073rz.rst:
```diff
@@ -0,0 +1,168 @@
+- ARM* mbed*
+- On-board ST-LINK/V2-1 debugger/programmer with SWD connector:
+
+    - Selection-mode switch to use the kit as a standalone ST-LINK/V2-1
```
ditto...

tagunil commented:
@dbkinder, I've fixed that and also image directives indentation. Is there any tool to check ReST formatting?

dbkinder commented:
Alas, no.  I haven't found a tool for checking ReST syntax and indenting (other than the Sphinx processor itself).

tagunil commented:
How about blank lines after sub-lists? Aren't they required?

dbkinder commented:
- Blank line before a list is required.  
- Blank lines between list items is optional. 
- An out-dent stops the list, making a blank line optional after the last item. (But it does make it more readable if a blank line is added after the last list item.)

tagunil commented:
Added these.

tagunil commented:
If this formatting is good enough, I'll make a PR reformatting other Nucleos soon.

tagunil commented:
recheck

dbkinder commented:
Yes. The .rst file looks great.

henrikbrixandersen commented:
recheck
